include/net/net_context.h

```diff
@@ -1076,6 +1091,19 @@ int net_context_set_option(struct net_context *context,
 			   enum net_context_option option,
 			   const void *value, size_t len);
 
+/**
+ * @brief Set tcp connection option for this context.
+ *
+ * @param context The network context to use.
+ * @param option Option to set
+ * @param value Option value
+ * @param len Option length
+ *
+ * @return 0 if ok, <0 if error
+ */
+int tcp_context_set_option(struct net_context *context,
```

Comment:
As this is part of `net_context` API, the preferrable name would be `net_context_tcp_set_option()`